src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -379,21 +382,25 @@ private Path createJNIConfig(String suffix) throws IOException {
         try (BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(f)))) {
             bw.write("[\n");
             bw.write("  {\n    \"name\" : \"" + projectConfiguration.getMainClassName() + "\"\n  }\n");
-            for (String javaClass : getJNIClassList(projectConfiguration.isUseJavaFX(), projectConfiguration.isUsePrismSW())) {
-                // TODO: create list of exclusions
-                if ("arm64-ios".equals(suffix) && javaClass.equals("java.lang.Thread")) {
-                    writeThreadEntry(bw);
-                } else {
-                    writeEntry(bw, javaClass);
-                }
+            for (String javaFile : getJNIClassList(projectConfiguration.isUseJavaFX(), projectConfiguration.isUsePrismSW())) {
+                bw.write(",\n");
+                Files.lines(Paths.get(AbstractTargetConfiguration.class.getResource("/config/" + javaFile).toURI()))
+                        .filter(line -> !line.startsWith("[") && !line.startsWith("]"))
+                        .forEach(line -> {
+                            try {
+                                bw.write(line + "\n");
+                            } catch (IOException e) {
+                                e.printStackTrace();
```

Comment:
probably should be fatal. If writing the JNI json file fails, there will be exceptions later (worst-time at runtime)